packages/ag-charts-enterprise/src/features/error-bar/errorBar.ts

```diff
@@ -297,25 +303,89 @@ export class ErrorBars
         const style = this.getDefaultStyle();
         node.datum = datum;
         node.updateStyle(style, this);
-        node.updateTranslation(this.cap, this.ctx.tooltipManager.getRange());
+        node.updateTranslation(this.cap);
+        node.updateBBoxes(this.getHighlightRange(), this.getNodeClickRange());
     }
 
-    private onHoverEvent(event: _ModuleSupport.InteractionEvent<'hover'>) {
+    private getHighlightRange() {
+        return this.ctx.tooltipManager.getRange();
+    }
+
+    private getNodeClickRange() {
+        return this.cartesianSeries.nodeClickRange;
+    }
+
+    private checkInteractionEventOverlap(event: InteractionEvent, range: InteractionRange): boolean {
+        if (range === 'nearest') {
+            return false;
+        }
+        // The error bars cover part of series datum node (marker/bar). If this part is clicked on, then
+        // that triggers the event twice (once in the Series class, and once in this ErrorBars class).
+        // We don't want that because there is an API for adding listeners to 'nodeClick' and
+        // 'nodeDoubleClick' events. Therefore, check whether the event hits a series node datum:
+        if (range === 'exact' || range === 0) {
+            const seriesNode = this.cartesianSeries.contentGroup.pickNode(event.offsetX, event.offsetY);
```

Comment:
`pickNode()` is a potentially expensive operation, we should avoid executing it multiple times (it's already in-play from a series perspective?) here and below.

Maybe we could consider 'disarming' the error-bar clickable state based upon the currently highlighted datum state?

Reply:
Yes, `pickNode()` is not ideal. However, we cannot use the highlight state to determine whether a node is clickable because we provide the option to customise the highlight and click range separately:
```
{
  tooltip: { range: 'exact' },
  series: [ { nodeClickRange: 10 ... } ],
}
```
This example makes it possible to click an unhighlighted node if the user clicks within a range of 10 pixels.

Essentially this function just needs to check whether the series node datum that the error bars map to is handling the event. So it doesn't need to do hit-testing for all of the nodes in `contentGroup`. We might be able to add a optimised method to just do hit-testing on a single a datum. Example:
```
private checkInteractionEventOverlap(datum: ErrorBarNodeDatum, event: InteractionEvent, range: InteractionRange): boolean {
    if (this.cartesianSeries.checkDatumHit(datum, event, range)) {
        return false;
    }
    return true;
}
```

Ideally, we want either the series datum node or the error bar node to consume the click event to avoid firing the `nodeClick` event twice. There is an `InteractionEvent.consume()` function. I tried calling that function in the _errorBar.ts_ file but it didn't prevent both nodes from firing the `nodeClick` event.

Reply:
Okay, so I've discovered that the `InteractionEvent.consume()` method is intended to suppress the event to other "internal" listeners like the legend and the error bars modules. The Series class could in theory consume this to hide it from all the other listener, but I don't think that this is desirable (it might unintentionally break things with the legend, for example).

Instead, I've added the `InteractionEvent.callbackComplete` property to keep track of whether or not the nodeClick / nodeDoubleClick have been broadcasted to the library user. https://github.com/ag-grid/ag-charts/commit/8193d3cdc36f2308da8121e1937069fc9b6aa8a6

I think that this is the best solution, because it avoids the need to recalculate the hit-tests and keeps a clear distinction between internal & external listeners.